pkg/config/config_template.yaml

```diff
@@ -413,6 +413,125 @@ api_key:
 #
 # secret_backend_timeout: 5
 
```

Comment:
It seems that we should have default values in https://github.com/DataDog/datadog-agent/blob/master/pkg/config/config.go

Instead of https://github.com/DataDog/datadog-agent/blob/967456455d1aca85ec67f2c057114aa8a284b2c4/pkg/snmp/snmp.go#L26-L28

And https://github.com/DataDog/datadog-agent/blob/967456455d1aca85ec67f2c057114aa8a284b2c4/pkg/autodiscovery/listeners/snmp.go#L25-L27

That would avoid code like

https://github.com/DataDog/datadog-agent/blob/967456455d1aca85ec67f2c057114aa8a284b2c4/pkg/autodiscovery/listeners/snmp.go#L211-L221

Probably better handled in another PR or new task in backlog.

Reply:
I don't think it's possible to have default values with the config system in embedded structures, which is why it's done this way. I'd be glad to be wrong though.

Reply:
Seems possible mapping configs like this: https://github.com/DataDog/datadog-agent/blob/d558b8335705918c94f807129f2fcf61c702ae4f/pkg/config/config.go#L414-L416

But yes, probably not possible for the config list of configs.